scripts/ci/vagrant.sh

```diff
@@ -49,4 +49,15 @@ fedora-no-vdso() {
 	ssh default 'cd /vagrant/criu/test; sudo ./zdtm.py run -a --keep-going'
 }
 
+fedora-non-root() {
+	ssh default 'curl -s https://repos.fedorapeople.org/repos/thl/kernel-vanilla.repo | sudo tee /etc/yum.repos.d/kernel-vanilla.repo'
+	ssh default 'sudo dnf -y --enablerepo=kernel-vanilla-mainline update kernel-core; sudo dnf -y install glibc-static'
+	vagrant reload
+	ssh default uname -a
+	ssh default 'cd /vagrant; tar xf criu.tar; cd criu; make -j 4'
+	ssh default 'cd /vagrant; git clone https://git.kernel.org/pub/scm/libs/libcap/libcap.git; cd libcap; make -C progs'
+	ssh default 'sudo /vagrant/libcap/progs/setcap cap_checkpoint_restore+eip /vagrant/criu/criu/criu'
+	ssh default 'cd /vagrant/criu; sudo ./test/zdtm.py run -t zdtm/static/env00 -f h --ignore-taint; sudo chmod 777 test test/zdtm/static/ test/dump/zdtm/static/env00; sudo ./test/zdtm.py run -t zdtm/static/env00 -f h --user --ignore-taint'
+}
+
```

Comment:
@adrianreber the Vagrant tests are currently not running in CI and it looks like the Ubuntu GitHub actions environment does not support nested virtualization (https://github.com/actions/virtual-environments/issues/183). However, it is possible to use macOS to run Vagrant VM (e.g. https://github.com/luxonis/depthai-docs-website/commit/f24e690c13c9d2fa5880b60fac2e327a527589d9).

Also, the Fedora 33 kernel provides CAP_CHECKPOINT_RESTORE.

Reply:
I have working vagrant using cirrus ci: #1272